modules/@apostrophecms/form-widget/ui/src/recaptcha.js

```diff
@@ -1,79 +1,50 @@
 /* global grecaptcha */
-let recaptchaSlot;
+let siteKey;
 let recaptchaError;
-let el;
 
 export default function (widgetEl) {
-  if (!widgetEl.querySelector('[data-apos-recaptcha-slot]')) {
+  if (!widgetEl.querySelector('[data-apos-recaptcha-sitekey]')) {
     return;
   }
 
-  recaptchaSlot = widgetEl.querySelector('[data-apos-recaptcha-slot]');
+  siteKey = widgetEl.querySelector('[data-apos-recaptcha-sitekey]').dataset.aposRecaptchaSitekey;
   recaptchaError = widgetEl.querySelector('[data-apos-form-recaptcha-error]');
-  el = widgetEl;
-
-  window.renderCaptchas = render;
 
   if (!document.querySelector('[data-apos-recaptcha-script]')) {
-    addRecaptchaScript();
+
+    window.enableSubmissions = function () {
+      grecaptcha.ready(function() {
+        const buttons = document.querySelectorAll('[data-apos-form-submit]');
+
+        Array.prototype.slice.call(buttons).forEach(btn => {
+          btn.disabled = false;
+        });
+      });
+    };
+    addRecaptchaScript(siteKey);
   }
 
   return {
-    getToken,
-    reset
+    getToken
   };
 }
 
-function addRecaptchaScript () {
+function addRecaptchaScript (siteKey) {
   const container = document.querySelector('[data-apos-refreshable]') || document.body;
   const recaptchaScript = document.createElement('script');
-  recaptchaScript.src = 'https://www.google.com/recaptcha/api.js?onload=renderCaptchas&render=explicit';
+  recaptchaScript.src = `https://www.google.com/recaptcha/api.js?render=${siteKey}&onload=enableSubmissions`;
   recaptchaScript.setAttribute('data-apos-recaptcha-script', '');
   recaptchaScript.setAttribute('async', '');
   recaptchaScript.setAttribute('defer', '');
   container.appendChild(recaptchaScript);
 }
 
-function getToken () {
-  const recaptchaId = recaptchaSlot.getAttribute('data-apos-recaptcha-id');
-  const token = grecaptcha.getResponse(recaptchaId);
-
-  if (!token) {
-    // They submitted without completing the reCaptcha
+async function getToken (widgetEl) {
+  try {
+    return grecaptcha.execute(siteKey, { action: 'submit' });
```

Comment:
This takes a half second. Could do it on page load and have the value waiting form them, but if the form is not present immediately that wouldn't work. There are probably other ways. Maybe more of this should be functional so people can use it as they want.

Reply:
It's a synchronous function that takes a half second? Seems like it just fails if the captcha isn't completed yet (this person's attempt to treat the return value as a promise doesn't work):

https://stackoverflow.com/questions/62015796/how-to-wait-for-the-recaptcha-v3-token-synchronously

But, here's someone who promisified it to make it awaitable, looks straightforward:

https://gist.github.com/luckylooke/847b61ced8e4bf706d5c2f3ac83d43fa

Am I misunderstanding? Not entirely sure where this half second is really coming into play.

Reply:
I'm not sure what you mean. There is documentation to use the execute method similarly here: https://developers.google.com/recaptcha/docs/v3#programmatically_invoke_the_challenge. Awaiting the response is working correctly.

Reply:
I see it returns an object with `then` and `catch` functions, not a traditional promise. It looks like await is meant to handle thenable objects like this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#thenable_objects